### ISSUE TITLE
Expose a way to know when all diagnostic requests for a project has been queued and processed

### ISSUE THREAD

**Issue by vladima** [dotnet/roslyn#12719]
In TypeScript scenarios it will allow us to promptly clean accumulated type information that otherwise will stay alive much longer increasing memory footprint.

Discussed it offline with @heejaechang - he asked to assign this issue to him

**Comment by heejaechang** [comment]
@vladima do you still need this?

**Comment by vladima** [comment]
yes, having this notifications will allow us to discard accumulated semantic data in multiproject scenarios for  projects that user have not touched

**Comment by CyrusNajmabadi** [comment]
Not planning on doing this.
